distribution/errors.go

```diff
@@ -60,6 +60,16 @@ func shouldV2Fallback(err errcode.Error) bool {
 	return false
 }
 
+type notFoundError struct {
```

Comment:
This is only used in one place. I think we can define it in a more specific way instead of wrapping a string.